src_nbgl/ui_sign_message.c

```diff
@@ -109,73 +111,53 @@ static bool nav_callback(uint8_t page, nbgl_pageContent_t *content) {
         }
     } else {
         // the last page must contain a long press button
-        display_sign(content);
+        ret = display_sign(content);
     }
     return ret;
 }
 
-static void choice_callback(bool confirm) {
-    if (confirm) {
-        nbgl_useCaseStatus("MESSAGE\nSIGNED", true, sign_message);
-        sign_message();
-    }
-}
-
 static void continue_review(void) {
-    nbgl_useCaseForwardOnlyReview("Reject", NULL, nav_callback, choice_callback);
+    nbgl_useCaseForwardOnlyReview(REJECT_BUTTON, NULL, nav_callback, ui_message_review_choice);
 }
 
-static void confirm_transaction_rejection_choice(bool confirm) {
-    if (confirm) {
-        reject_message();
+static void resume_message(void) {
+    e_ui_191_action action_bak = g_action;
+
+    if (g_position == UI_SIGNING_POSITION_START) {
+        ui_191_start();
     } else {
-        // Go to previous screen accordingly
-        if (state == UI_191_NBGL_START_REVIEW_DISPLAYED) {
-            ui_191_start();
-        } else {
-            if (state != UI_191_NBGL_SIGN_DISPLAYED) {
-                state_before_reject_cancel = state;
-                state = UI_191_NBGL_BACK_FROM_REJECT_CANCEL;
-            }
-            continue_review();
-        }
+        g_action = UI_191_ACTION_IDLE;
+        continue_review();
+        g_action = action_bak;
     }
 }
 
-static void confirm_transaction_rejection() {
-    nbgl_useCaseChoice(&C_warning64px,
-                       "Reject message?",
-                       NULL,
-                       "Yes, Reject",
-                       "Go back to message",
-                       confirm_transaction_rejection_choice);
-}
-
 void ui_191_start(void) {
-    state = UI_191_NBGL_START_REVIEW_DISPLAYED;
+    g_position = UI_SIGNING_POSITION_START;
+
     skip_message = false;
     eip191MessageIdx = 0;
     stringsTmpTmpIdx = 0;
 
-    nbgl_useCaseReviewStart(&C_Message_64px,
-                            "Review message",
-                            NULL,
-                            "Reject",
-                            continue_review,
-                            confirm_transaction_rejection);
+    ui_message_start(REVIEW(MESSAGE_EIP191),
```

Comment:
I feel like having an explicit `EIP191_MESSAGE_REVIEW_HEADER_TEXT` is more clear than function macros
function macros don't even gain space as it's expanded at compilation anyway.
Also, putting all NBGL texts in a single file `nbgl_localization.h` would be very clear

Even if we decide we want to use the macro function, it could/should be hidden in an internal file
eg
```
#define EIP191_MESSAGE_REVIEW_HEADER_TEXT REVIEW(MESSAGE_EIP191)
```
And the user function uses `EIP191_MESSAGE_REVIEW_HEADER_TEXT`

Reply:
Indeed, I added some intermediate macros. :+1: